fipy/matrices/pysparseMatrix.py

```diff
@@ -403,18 +405,24 @@ def __init__(self, rows, cols, bandwidth=0, sizeHint=None, matrix=None, storeZer
             The number of matrix rows
         cols : int
             The number of matrix columns
-        bandwidth : int
-            The proposed band width of the matrix.
-        sizeHint : int
-            Estimate of the number of non-zeros
+        nonZerosPerRow : int or array_like of int
+            The approximate number of sparse entries per row.  Either a
+            typical number, or an iterable of values for each row
+            (default: 0).
+        exactNonZeros : bool
+            *ignored*
         matrix : ~pysparse.spmatrix.ll_mat
             Pre-assembled Pysparse matrix to use for storage
         storeZeros : bool
             Instructs pysparse to store zero values if possible.
         """
-        sizeHint = sizeHint or max(rows, cols) * bandwidth
         if matrix is None:
             tmpMatrix = spmatrix.ll_mat(1, 1, 1)
+            try:
```

Comment:
Can you do this without a try / except? It just feels wrong here. Does the `asarray` call raise the `TypeError`?

Reply:
The `len(nonZerosPerRow)` is what throws the `TypeError`. This is the most effective way to figure out if `nonZerosPerRow` is iterable in Python 2.7. In Py3k, you can (and I do) do an `isinstance(nonZerosPerRow, collections.abc.Iterable)`.

Reply:
Oh yeah, FiPy still needs to work with 2.7.